packages/expo-cli/src/commands/upgrade.ts

```
@@ -0,0 +1,217 @@
+  let bundledNativeModules = (await JsonFile.readAsync(
+    ConfigUtils.resolveModule('expo/bundledNativeModules.json', projectRoot, exp)
+  )) as DependencyList;
+
+  let { dependencies } = pkg;
+  Object.keys(bundledNativeModules).forEach(name => {
+    if (dependencies[name]) {
+      result[name] = bundledNativeModules[name];
+    }
+  });
+
+  return result;
+}
+
+async function upgradeAsync(requestedSdkVersion: string | null, options: Options) {
+  let { projectRoot, workflow } = await findProjectRootAsync(process.cwd());
+  let { exp } = await ConfigUtils.readConfigJsonAsync(projectRoot);
+
+  if (Versions.lteSdkVersion(exp, '32.0.0')) {
+    log.error('The upgrade command is only available on SDK 33 and higher. Please refer to https://docs.expo.io/versions/latest/workflow/upgrading-expo-sdk-walkthrough/ for upgrade instructions.');
+    return;
+  }
+
+  // Can't upgrade if we don't have a SDK version
+  if (!exp.sdkVersion) {
+    log.error('No sdkVersion field is present in app.json, cannot upgrade project.');
+    return;
+  }
+
+  let currentSdkVersionString = exp.sdkVersion;
+  let sdkVersions = await Versions.sdkVersionsAsync();
+  let latestSdkVersion = await Versions.newestSdkVersionAsync();
+  let latestSdkVersionString = latestSdkVersion.version;
+  let targetSdkVersionString =
+    maybeFormatSdkVersion(requestedSdkVersion) || latestSdkVersion.version;
+  let targetSdkVersion = sdkVersions[targetSdkVersionString];
+
+  // TODO: figure out how this should work for bare workflow
+  if (workflow === 'bare') {
+    console.log('This command is currently only supported on the managed workflow.');
+    return;
+  }
+
+  // Bail out early if people are trying to update to the current version
+  if (targetSdkVersionString === currentSdkVersionString) {
+    console.log(
+      'You are already using the latest SDK version. Follow https://blog.expo.io for new release information.'
+    );
+    return;
```

We should throw an error (e.g. `CommandError`) to make the return value of the command non-zero.

---

i think in this case i'd expect it to exit with status 0 because, while it didn't upgrade, the command ran successfully. we already were on latest version so no need to change anything